presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/mysql/FunctionNamespaceDao.java

```diff
@@ -51,13 +51,14 @@
 
     @SqlUpdate("CREATE TABLE IF NOT EXISTS <sql_functions_table> (\n" +
             "  id bigint(20) NOT NULL AUTO_INCREMENT,\n" +
-            "  function_id varchar(700) NOT NULL,\n" +
+            "  function_id_hash varchar(128) NOT NULL,\n" +
+            "  function_id text NOT NULL,\n" +
             "  version bigint(20) unsigned NOT NULL,\n" +
             "  catalog_name varchar(128) NOT NULL,\n" +
             "  schema_name varchar(128) NOT NULL,\n" +
             "  function_name varchar(256) NOT NULL,\n" +
-            "  parameters varchar(40000) NOT NULL,\n" +
-            "  return_type varchar(256) NOT NULL,\n" +
+            "  parameters mediumtext NOT NULL,\n" +
```

Comment:
text are stored in separate pages form the table so moving this to text effectively adds an additional page load to each row. Overall text type is very inefficient. I think this is an overkill as a solution to support large parameters. Most of the functions would have a small parameter list. I suggest to use encoding rather than just dump everything blindly into a blob.

Reply:
Can you elaborate the encoding you are mentioning. What does it mean? Do you mean String compression, or a more efficient encoding such as binary?

Currently, we're already using "encoding" - JSON encoding. What are the other types of encoding that we can use to reduce the size of this field while avoid using text? Are there any standard out there? For example, I can think thrift binary encoding as an alternative, but that will require a schema IDL to be kept elsewhere, which is not convenient. Also, only applications will be able to read the data, making debugging and human inspection difficult.

It's unclear to me what encoding can be used and if there is guaranteed size shrink that can be achieved by using that. Also, please note that each MySQL table row has a max size of 65536 bytes. Potentially, if we avoid text by using large columns, say varchar(15000), that will limit the number of those large varchar column we can support.

I would say that the performance increase in using text is very minimal since all of our queries are backed by index. Point reads are instantaneous and have minimal cost. The only feature that requires reading a lot of rows is `listFunctions`, in which reads will still be instantaneous when we have less than 10^5 records, which is still sufficient in the vast majority of the use cases.

There is always a limit for any systems, I would say MySQL-back Function Namespace Manager would not be a right tool to use, not just for this particular column, if we have more than, say 10^7 records.

Reply:
It's not about encoding. It's more about InnoDB internals. Let's chat over VC. I can give you a 101 on Innodb. 😂